protected_tags.go

```diff
@@ -0,0 +1,141 @@
+package gitlab
+
+import (
+	"fmt"
+	"net/url"
+)
+
+// ProtectedTagsService handles communication with the protected tag methods of the GitLab API.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html
+type ProtectedTagsService struct {
+	client *Client
+}
+
+// TagAccessDescription reperesents the access decription for a protected tag.
+//
+// GitLab API docs:
+// https://docs.gitlab.com/ee/api/protected_tags.html#list-protected-tags
+type TagAccessDescription struct {
+	AccessLevel            AccessLevelValue `json:"access_level"`
+	AccessLevelDescription string           `json:"access_level_description"`
+}
```

Comment:
I'm not sure if there is much value in defining this as a specific type. So I prefer to add these 2 fields directly to `ProtectedTag` using an inline struct.

Reply:
Thanks for the feedback! Moving `TagAccessDescription` into an anonymous struct inside `ProtectedTag` requires adding some duplicate definition in the unit tests due to how you define anonymous structs:

```
type ProtectedTag struct {
	Name               string `json:"name"`
	CreateAccessLevels []struct {
		AccessLevel            AccessLevelValue `json:"access_level"`
		AccessLevelDescription string           `json:"access_level_description"`
	} `json:"create_access_levels"`
}
```

Example updated unit test:
```
func TestListProtectedTags(t *testing.T) {
	mux, server, client := setup()
	defer teardown(server)

	mux.HandleFunc("/api/v4/projects/1/protected_tags", func(w http.ResponseWriter, r *http.Request) {
		testMethod(t, r, "GET")
		fmt.Fprint(w, `[{"name":"1.0.0", "create_access_levels": [{"access_level": 40, "access_level_description": "Maintainers"}]},{"name":"*-release", "create_access_levels": [{"access_level": 30, "access_level_description": "Developers + Maintainers"}]}]`)
	})

	expected := []*ProtectedTag{
		{
			Name: "1.0.0",
			CreateAccessLevels: []struct {
				AccessLevel            AccessLevelValue `json:"access_level"`
				AccessLevelDescription string           `json:"access_level_description"`
			}{
				{
					AccessLevel:            40,
					AccessLevelDescription: "Maintainers",
				},
			},
		},
		{
			Name: "*-release",
			CreateAccessLevels: []struct {
				AccessLevel            AccessLevelValue `json:"access_level"`
				AccessLevelDescription string           `json:"access_level_description"`
			}{
				{
					AccessLevel:            30,
					AccessLevelDescription: "Developers + Maintainers",
				},
			},
		},
	}

	opt := &ListProtectedTagsOptions{}
	tags, _, err := client.ProtectedTags.ListProtectedTags(1, opt)
	assert.NoError(t, err, "failed to get response")
	assert.Equal(t, expected, tags)
}
``` 

Is this what you mean by an inline struct? If so, and if this duplication in the tests is fine, I don't mind making this change.